Source/WTF/wtf/JSONValues.cpp

```diff
@@ -460,49 +461,61 @@ RefPtr<JSON::Value> buildValue(const CodeUnit* start, const CodeUnit* end, const
 
 } // anonymous namespace
 
-Ref<Value> Value::null()
+template<typename Visitor> constexpr decltype(auto) Value::visitDerived(Visitor&& visitor)
 {
-    return adoptRef(*new Value);
+    switch (m_type) {
+    case Type::Null:
+    case Type::Boolean:
+    case Type::Double:
+    case Type::Integer:
+    case Type::String:
+        return std::invoke(std::forward<Visitor>(visitor), static_cast<Value&>(*this));
```

Comment:
NIT: is the `static_cast` actually needed here since it's a member function of `Value`?  or is the thinking that we'd wanna `static_cast` anyways if we're operating on a subclass?

Reply:
This is not necessary because this is the same class, but I think this is good-to-have to align to the other things (so, we can always use this form even if we add yet another type here).